agent/bench-scripts/pbench-fio

```diff
@@ -567,22 +567,34 @@ function fio_run_job() {
 	pbench-start-tools --group=$tool_group --iteration=$iteration --dir=$benchmark_results_dir
 	local client_opts=""
 
-	if [ ! -z "$client_file" ] ;then
-		typeset -i nclients=$(wc -l $client_file | cut -d ' ' -f 1)
-		client_opts="--client=$client_file --max-jobs=$nclients"
-	elif [ ! -z "$clients" ]; then
+	# if a client-file parameter was passed, then clients is already 
+	# initialized to its contents
+
+	if [ ! -z "$clients" ]; then
 		local max_jobs=0
+		local nclients=0
+
+		if [ ! -z $numjobs ]; then
+			# so that a separate filename will be generated
+			# for each (host, job-number) pair
+			let max_jobs=$numjobs
+		else
+		       for t in `echo $targets | sed -e s/,/" "/g` ; do
+			       let max_jobs=$max_jobs+1
+		       done
+		fi
+
+		# make the fio command shorter by always using a client file
+		# regardless of how clients were specified
+		# we always save the client list in the benchmark results directory
+
+		client_file=$benchmark_results_dir/pbench-clients.list
```

Comment:
If `$client_file` is already defined, won't this override that global variable?  If so, can we add a comment saying that we are explicitly doing that?

Also, why re-write the file with the same contents?

Reply:
"won't this override that global variable":  I suppose you would have preferred if I defined a new variable.   Context: I was trying to unify the handling of --client and --client-file.    When --client-file is parsed, the "clients" variable is initialized with the contents of the file.   So from then on we just deal with the "clients" variable.    But I didn't like running fio with each client passed individually on the command line, this makes fio command *very* long in the case of a large-scale test, so I decided to create a file containing the list of clients in the benchmark directory (so it is saved with test results), and do fio --client=$client_file instead.    It's very important that the contents of client_file be saved in the results directory since otherwise we have no record of this input to fio.

"why rewrite the file": So you wanted to copy the pathname in $client_file to $benchmark_results_dir/pbench-clients.list instead, but that directory didn't exist when the --client-file option was parsed, and the way I'm doing it correctly uses the contents of that file and saves the list of client host/ips in the results directory, so why not?

"rebase since it won't merge as is": perhaps it won't because of the regression test failures, but prior to today github was telling me that it was ok to merge it.    I'll look at the test failures and see what's going on today.  I'm reluctant to rebase (because this requires a complete retest) unless there is a good reason for it (i.e. git can't merge the code), again I'll take a look and see if that is the case.